instrumentation/pulsar/pulsar-2.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pulsar/v2_8/telemetry/PulsarSingletons.java

```diff
@@ -151,6 +153,22 @@ private static Instrumenter<PulsarRequest, Void> createProducerInstrumenter() {
     return builder.buildProducerInstrumenter(MessageTextMapSetter.INSTANCE);
   }
 
+  private static Instrumenter<String, Void> createTxnProduceInstrumenter() {
+    InstrumenterBuilder<String, Void> instrumenterBuilder =
+        Instrumenter.builder(
+            TELEMETRY, INSTRUMENTATION_NAME, request -> "Txn Produce Register Topic");
+    return instrumenterBuilder.buildInstrumenter(SpanKindExtractor.alwaysProducer());
+  }
+
+  public static Context startAndEndTxnProduceRegister(
+      Context parent, Timer timer, Throwable throwable) {
+    if (!TXN_PRODUCE_INSTRUMENTER.shouldStart(parent, "")) {
+      return null;
+    }
+    return InstrumenterUtil.startAndEnd(
+        TXN_PRODUCE_INSTRUMENTER, parent, "", null, throwable, timer.startTime(), timer.now());
+  }
```

Comment:
My advice would be not to generate this span. Such spans need to follow the semantic conventions. You'd first need to work with the specification to describe when such nested (the test expects nested, but the code generates sibling, Idk which you really wanted) spans should be used.

Reply:
I am also hesitating whether to generate this span, maybe don't generate it is better.